vm/systemSmartContracts/delegationManager.go

```diff
@@ -97,29 +113,262 @@ func (d *delegationManager) Execute(args *vmcommon.ContractCallInput) vmcommon.R
+	}
+
+	d.eei.Finish(newAddress)
+
 	return vmcommon.Ok
 }
+func (d *delegationManager) checkConfigChangeInput(args *vmcommon.ContractCallInput) error {
+	if args.CallValue.Cmp(zero) != 0 {
+		return vm.ErrCallValueMustBeZero
+	}
+	if len(args.Arguments) != 1 {
+		return vm.ErrInvalidNumOfArguments
+	}
+	if !bytes.Equal(args.CallerAddr, d.delegationMgrSCAddress) {
+		return vm.ErrInvalidCaller
+	}
+	return nil
+}
 
-func (d *delegationManager) createNewDelegationContract(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	return vmcommon.Ok
+func (d *delegationManager) changeBaseIssuingCost(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	err := d.checkConfigChangeInput(args)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	delegationManagment, err := d.getDelegationManagementData()
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	baseIssuingCost := big.NewInt(0).SetBytes(args.Arguments[0])
+	if baseIssuingCost.Cmp(zero) < 0 {
+		d.eei.AddReturnMessage("invalid base issuing cost")
+		return vmcommon.UserError
+	}
+	delegationManagment.BaseIssueingCost = baseIssuingCost
+	err = d.saveDelegationManagementData(delegationManagment)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	return vmcommon.UserError
 }
 
-func (d *delegationManager) init(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	return vmcommon.Ok
+func (d *delegationManager) changeMinDeposit(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	err := d.checkConfigChangeInput(args)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	delegationManagment, err := d.getDelegationManagementData()
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	minDeposit := big.NewInt(0).SetBytes(args.Arguments[0])
+	if minDeposit.Cmp(zero) < 0 {
+		d.eei.AddReturnMessage("invalid base issuing cost")
+		return vmcommon.UserError
+	}
+	delegationManagment.MinDeposit = minDeposit
+	err = d.saveDelegationManagementData(delegationManagment)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	return vmcommon.UserError
+}
+
+func (d *delegationManager) getAllContractAddresses(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !bytes.Equal(args.CallerAddr, d.delegationMgrSCAddress) {
+		d.eei.AddReturnMessage(vm.ErrInvalidCaller.Error())
+		return vmcommon.UserError
+	}
+
+	contractList, err := d.getDelegationContractList()
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	for _, address := range contractList.Addresses {
+		d.eei.Finish(address)
+	}
+
+	return vmcommon.UserError
+}
+
+func createNewAddress(lastAddress []byte) []byte {
+	newAddress := make([]byte, 0, len(lastAddress))
+	copy(newAddress, lastAddress)
+
+	for i := len(newAddress) - 1; i > 0; i++ {
```

Comment:
I do not think this is a good approach as this will consume the entire space of addresses in a few thousands calls.

Reply:
few trillions of calls.

Reply:
this allocates the next one always. there are not enough funds to take up all the addresses.